src/runtime_src/xdp/profile/plugin/device_offload/hal/hal_device_offload_plugin.cpp

```diff
@@ -96,16 +69,58 @@ namespace xdp {
     }
   }
 
+  void HALDeviceOffloadPlugin::init()
+  {
+    if (initialized) {
+      return;
+    }
+    // Open all existing devices so that XDP can access the owned handles
+    uint32_t numDevices = xrt_core::get_total_devices(true).second;
+    uint32_t index = 0;
+    while (index < numDevices) {
+      try {
+        xrtDevices.push_back(std::make_unique<xrt::device>(index));
+
+        auto ownedHandle = xrtDevices[index]->get_handle()->get_device_handle();
+        std::string path = util::getDebugIpLayoutPath(ownedHandle);
+
+        if ("" != path) {
+          // TODO: Update this code to use device ID from the database 
+          uint64_t deviceId = addDevice(path);
```

Comment:
We are two ways to get "deviceId" now. Once here and then using getDeviceContextUniqueId later. I understand that the initialization is required for Alveo flow to identify all available boards. Please use VPStaticDatabase::getDeviceContextUniqueId here.